libobs-d3d11/d3d11-subsystem.cpp

```diff
@@ -1326,6 +1332,16 @@ static inline void LogD3DAdapters()
 	if (FAILED(hr))
 		throw HRError("Failed to create DXGIFactory", hr);
 
+	/* D3DKMT has its own handles that we need to enumerate */
+	d3dkmt_enum.NumAdapters = MAX_ADAPTERS;
+	d3dkmt_enum.pAdapters = d3dkmt_infos;
+	res = D3DKMTEnumAdapters2(&d3dkmt_enum);
+	if (FAILED(res)) {
+		blog(LOG_INFO,
+		     "D3DKMTEnumAdapters2 failed, HAGS information will be unavailable");
+		d3dkmt_enum.NumAdapters = 0;
+	}
```

Comment:
Do we know why it could fail? I suppose we might just have to see how often this fails in the wild, if ever.

Reply:
In theory it would fail if the user has more than 16 GPUs. But like with most Win32 API usage, it's checked here "just in case".

Reply:
Good to know.